src/modules/formBuilder/components/itemEditor/conditionals/AutofillValueCard.tsx

```diff
@@ -0,0 +1,107 @@
+import ControlledTextInput from '@/modules/form/components/rhf/ControlledTextInput';
+import { ItemMap } from '@/modules/form/types';
+import { ItemType } from '@/types/gqlTypes';
+
+interface AutofillValueCardProps {
+  control: FormItemControl;
+  index: number;
+  itemMap: ItemMap;
+  title: string;
+  itemType: ItemType; // will be used to determine value type
+}
+
+// Card for managing a single AutofillValue
+const AutofillValueCard: React.FC<AutofillValueCardProps> = ({
+  control,
+  index,
+  itemMap,
+  title,
+}) => {
+  //TODO: also accept sum_questions for autofilling numeric fields using a sum of other questions
+
+  return (
+    <>
+      <Typography sx={{ mb: 2 }}>{title}</Typography>
+
+      <Stack gap={2}>
+        <Typography variant='body2'>
+          Enter the value to autofill, or enter a formula to calculate it:
+        </Typography>
+
+        <ControlledTextInput
+          name={`autofillValues.${index}.valueCode`}
+          label='Value (String / Code)'
+          control={control}
+          // disabled={[
+          //   ItemType.Boolean,
+          //   ItemType.Currency,
+          //   ItemType.Integer,
+          // ].includes(itemType)}
+        />
+        <Controller
+          name={`autofillValues.${index}.valueBoolean`}
+          control={control}
+          // disabled={itemType !== ItemType.Boolean}
+          render={({
+            field: { ref, disabled, ...field },
+            fieldState: { error },
+          }) => (
+            <YesNoRadio
+              label='Value (Boolean)'
+              sx={disabled ? { display: 'none' } : {}}
+              error={!!error}
+              helperText={error?.message}
+              {...field}
+            />
+          )}
+        />
+        <Controller
+          name={`autofillValues.${index}.valueNumber`}
+          control={control}
+          // disabled={![ItemType.Currency, ItemType.Integer].includes(itemType)}
+          render={({
+            field: { ref, disabled, ...field },
+            fieldState: { error },
+          }) => (
+            <NumberInput
+              sx={disabled ? { display: 'none' } : undefined}
+              label='Value (Numeric)'
+              inputRef={ref}
+              error={!!error}
+              helperText={error?.message}
+              {...field}
+            />
+          )}
+        />
+        <ControlledTextInput
+          name={`autofillValues.${index}.formula`}
+          control={control}
+          // TODO: validate formula
+          // disabled={!answerInputTypes.includes('answerCode')}
+          label='Value (Formula)'
+          helperText="Formula to calculate the value to fill. Use 'value' to refer to the value of the current item."
+        />
+        <ControlledCheckbox
+          name={`autofillValues.${index}.autofillReadonly`}
+          control={control}
+          label='Autofill in read-only mode'
+        />
+        <ManageEnableWhen
```

Comment:
I'm not sure of the right place for this, but I think it could be helpful to add some comments about the overlap between Autofill and EnableWhen. My understanding is that we use the ManageEnableWhen component for Autofill logic because we are trying to do two similar things here -- indicate a set of conditions + when those conditions are applicable -- and so the data shape is the same. And in the [json form schema](https://github.com/greenriver/hmis-warehouse/blob/release-118/drivers/hmis_external_apis/public/schemas/form_definition.json#L181) itself we also reuse EnableWhen for Autofill. Maybe commentary about this already exists somewhere?

Reply:
That's right. I'll add some comments on these components explaining that the "EnableWhen" object type is used in two places in the schema (conditional rules for visibility + conditional rules for autofill).

There is a brief comment here https://github.com/greenriver/hmis-warehouse/blob/e4d8e97cbb875889094457e18178cb515692c126/drivers/hmis/app/graphql/types/forms/autofill_value.rb#L22-L24